### PR TITLE
Jobservice control loop (based on #1140)

### PR DESCRIPTION
**Pull request by oavdeev**
**What this PR does / why we need it**:
See #1140 for start of the discussion. Compared to that PR, this addresses some comments and adds tests for the control loop logic.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: sdk/python/tests/test_streaming_control_loop.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+    try:
+        address = "localhost:%d" % port
+        with grpc.insecure_channel(address):
+            yield address
+    finally:
+        server.stop(None)
+
+
+CORE_URL = "core.feast.example.com"
```
Comment (woop) [inline sdk/python/tests/test_streaming_control_loop.py:40]: Do we need these?

File: sdk/python/tests/test_streaming_control_loop.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            # We expect a streaming job to be created for the new Feature Table.
+            assert len(client.list_jobs(include_terminated=False)) == 1
+            assert len(processes) == 1
+
+            first_job_id = client.list_jobs(include_terminated=False)[0].get_id()
+
+            # Pretend that the streaming job has terminated for no reason.
+            processes[0].kill()
+
+            # The control loop is expected to notice the killed job and start it again.
```
Comment (woop) [inline sdk/python/tests/test_streaming_control_loop.py:150]: Should we perhaps add one assert here to test that a job has terminated?

File: sdk/python/tests/test_streaming_control_loop.py (hunk 1 above)
Comment (oavdeev-tt) [reply sdk/python/tests/test_streaming_control_loop.py:40]: You're right we don't

File: sdk/python/tests/test_streaming_control_loop.py (hunk 2 above)
Comment (oavdeev-tt) [reply sdk/python/tests/test_streaming_control_loop.py:150]: Hmm not sure what code path would that help test. We just terminated it one line above this so we know it is dead. The assert below checks that there is one terminated and one started job after we ran `ensure_stream_ingestion_jobs`

Comment (woop) [reply sdk/python/tests/test_streaming_control_loop.py:150]: Fair enough.

**Review by woop — APPROVED** [review]

**Pull request merged**